### PR TITLE
Replace usages of `global` with `globalThis`

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes https://github.com/matrix-org/matrix-js-sdk/issues/4450

This should solve the problem with various bundlers (Vite etc) not being happy about the usage of `global`, with `globalThis`. All modern environments support this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis, so it's a safe change.

### CODE REVIEW

**Comment by t3chguy** [comment]
Any chance of a linter to protect against regression here?

**Comment by Half-Shot** [comment]
> Any chance of a linter to protect against regression here?

Good idea! Let's have a look.

File: spec/unit/pushprocessor.spec.ts [hunk 1]
```diff
@@ -819,21 +819,21 @@ describe("rewriteDefaultRules", () => {
 
         // By the time we get here, we expect the PushProcessor to have merged the new .m.rule.is_room_mention rule into the existing list of rules.
         // Check that has happened, and that it is in the right place.
-        const containsDisplayNameRuleIdx = pushRules.global.override?.findIndex(
+        const containsDisplayNameRuleIdx = pushRules.globalThis.override?.findIndex(
```
Comment (t3chguy) [inline spec/unit/pushprocessor.spec.ts:822]: 🙃

Comment (Half-Shot) [reply spec/unit/pushprocessor.spec.ts]: Believe it or not I did actually do these replaces manually, and still fucked it up :grimacing:

**Review by t3chguy — APPROVED** [review]

**Review by richvdh — APPROVED** [review]
lgtm

**Review by fkwp — APPROVED** [review]
lgtm

**Comment by Half-Shot** [comment]
(N.B. leaving branch active for the moment as we're depending on it to unbreak some applications, once the *next* RC comes out this can be pruned)

**Pull request merged**